Sources/GRPCProtobuf/Errors/GoogleRPCStatus.swift

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2024, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+public import GRPCCore
+internal import SwiftProtobuf
+
+/// An error containing structured details which can be delivered to the client.
+///
+/// This error follows the "richer error model" detailed in the
+/// [gRPC error guide](https://grpc.io/docs/guides/error/) and
+/// [Google AIP-193](https://google.aip.dev/193).
+///
+/// Like an `RPCError`, this error has a `code` and `message`. However it also includes
+/// a list of structured error details which can be propagated to clients. A set of standard
+/// details are provided by ``ErrorDetails``.
+///
+/// As a client you can extract this error from an `RPCError` using `unpackGoogleRPCStatus()`.
+///
+/// > Implementation details:
+/// >
+/// > The error information is transmitted to clients in the trailing metadata of an RPC. It is
+/// > inserted into the metadata keyed by "grpc-status-details-bin". The value of the metadata is
+/// > the serialized bytes of a "google.protobuf.Any" protocol buffers message. The content of which
+/// > is a "google.rpc.Status" protocol buffers message containing the status code, message, and
+/// > details.
+public struct GoogleRPCStatus: Error {
```

Comment:
Why not `GoogleRPCError`? We refer to it as an error, conforms to `Error`, we compare it to `RPCError`, and it wraps an `RPCError.Code` and `[ErrorDetails]`.

Reply:
Naming this type is annoying... I picked `GoogleRPCStatus` because it's effectively a "google.rpc.Status". 

Other implementations which support this use the generated proto directly so also call it similarly (depends a bit on the language and how they generate proto messages). I elected _not_ do that for a number of reasons; naming, retroactive conformance to `RPCErrorConvertible`, user experience etc.

The other idea I had for naming it was `DetailedRPCError`.

Reply:
I think there's a lot of benefit to giving it a name thats similar to other implementations especially as this _isn't_ a gRPC feature, it's a widely used convention built on top of gRPC. 

If this was a custom grpc-swift thing then I'd definitely prefer a name like `DetailedRPCError`.